src/main/java/com/microsoft/jenkins/appservice/WebAppSwapSlotsRecorder.java

```diff
@@ -0,0 +1,190 @@
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
+public class WebAppSwapSlotsRecorder extends Recorder implements SimpleBuildStep {
+    private final String azureCredentialsId;
+    private final String resourceGroup;
+    private final String appName;
+    private final String sourceSlotName;
+    private final String targetSlotName;
+
+    @DataBoundConstructor
+    public WebAppSwapSlotsRecorder(String azureCredentialsId,
+                                   String resourceGroup,
+                                   String appName,
+                                   String sourceSlotName,
+                                   String targetSlotName) {
+        this.azureCredentialsId = azureCredentialsId;
+        this.resourceGroup = resourceGroup;
+        this.appName = appName;
+        this.sourceSlotName = sourceSlotName;
+        this.targetSlotName = targetSlotName;
+    }
+
+    public String getAzureCredentialsId() {
+        return azureCredentialsId;
+    }
+
+    public String getResourceGroup() {
+        return resourceGroup;
+    }
+
+    public String getAppName() {
+        return appName;
+    }
+
+    public String getSourceSlotName() {
+        return sourceSlotName;
+    }
+
+    public String getTargetSlotName() {
+        return targetSlotName;
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace,
+                        @Nonnull Launcher launcher, @Nonnull TaskListener listener)
+            throws InterruptedException, IOException {
+
+        final Azure azureClient = AzureUtils.buildClient(run.getParent(), azureCredentialsId);
+        WebApp app;
+        try {
+            app = azureClient.webApps().getByResourceGroup(resourceGroup, appName);
+        } catch (NoSuchElementException e) {
+            throw new AbortException(String.format("Web App %s in resource group %s not found",
+                    appName, resourceGroup));
+        }
+        if (app == null) {
+            throw new AbortException(String.format("Web App %s in resource group %s not found",
+                    appName, resourceGroup));
+        }
+
+        WebAppSwapSlotsCommandContext context = new WebAppSwapSlotsCommandContext();
+        context.setSourceSlotName(sourceSlotName);
+        context.setTargetSlotName(targetSlotName);
+        context.setWebApp(app);
+
+        try {
+            context.configure(run, workspace, launcher, listener, app);
+        } catch (AzureCloudException e) {
+            throw new AbortException(e.getMessage());
+        }
+
+        context.executeCommands();
+
+        if (!context.getLastCommandState().isError()) {
+            listener.getLogger().println(String.format("Done swapping Azure Web "
+                    + "App deployment slots between %s and %s.", sourceSlotName, targetSlotName));
+        } else {
+            throw new AbortException("Azure Web App swapping slots failed.");
```

Comment:
add exception detail?

Reply:
The exception detail has already been recorded in the executed command. In the current design, no error message is available at this context.